uswds/uswds-gulp
# Update task composition.

## About
Resolves #38. This PR aims to modernize our gulp tasks and improve the developer experience.

### Available tasks
- watch
- copySetup
- copyFonts
- copyImages
- copyJS
- copyAll `[copySetup, copyFonts, copyImages, copyJS]`
- buildSass
- buildSvgSprite
- init `[copyAll, buildSass]`
- default `[init]`

#### Running tasks
There are three different ways to set this up and run tasks:

- Copy gulpfile and run tasks as needed 🌟
- Import tasks and customize as needed
- Run tasks directly

🌟 The preferred way

## How to test
1. Pull down `jm-gulp-4` branch.
2.  Run `npm i`
3. Run `npm link`
4. Go to a test USWDS project and run `npm link uswds-gulp`
5. Copy gulpfile, browserlistrc, and run tasks
6. Test creating a custom `gulpfile.js` and importing tasks
7. Test running tasks directly from uswds-gulp (without copying)

All tasks should run and compile successfully. You can find additional details in [README](https://github.com/uswds/uswds-gulp/blob/jm-gulp-4/README.md)
